embedded-graphics/src/primitives/line.rs

```diff
@@ -156,43 +203,20 @@ where
 {
     style: PrimitiveStyle<C>,
 
-    start: Point,
-    end: Point,
-    delta: Point,
-    /// in which quadrant is the line drawn (upper-left=(-1, -1), lower-right=(1, 1), ...)
-    direction: Point,
-    err: i32,
-    stop: bool,
+    line_iter: LineIterator,
 }
 
 // [Bresenham's line algorithm](https://en.wikipedia.org/wiki/Bresenham%27s_line_algorithm)
 impl<C: PixelColor> Iterator for StyledLineIterator<C> {
     type Item = Pixel<C>;
 
     fn next(&mut self) -> Option<Self::Item> {
-        // return none if stroke color is none
+        // Return none if stroke color is none
         let stroke_color = self.style.stroke_color?;
```

Comment:
This problem probably existed before this refactoring, but to be consistent with rectangles and circles, lines should also not be drawn if `stroke_width` is `0`.

This also applies to triangles. Setting one of the stroke width of one of the styles in the `stroke.rs` example to `0` is a nice demo of this problem.

Reply:
Good catch! This also uncovered another problem with triangles and their borders which we haven't uncovered until now. All the triangle demos/tests use `BinaryColor` which masks this overdraw bug:

![image](https://user-images.githubusercontent.com/800791/75437755-a8a61800-594e-11ea-9cc0-10e06696495c.png)

I've pushed some half-done debugging to this branch which I'll finish off later.

Reply:
And another example: the top-left line is overlapped by the border.

![Screenshot from 2020-02-27 14-36-57](https://user-images.githubusercontent.com/800791/75455266-309d1980-5971-11ea-87ea-afe37dc293d9.png)

Reply:
Also a good catch!